### PR TITLE
Ref 427 dac hint usability 01

### PR DESCRIPTION
**Pull request by elceebee**
This is a recommendation coming from the referrals accessibility report. 

https://citizensadvice.atlassian.net/browse/REF-427

### CODE REVIEW

File: engine/spec/components/citizens_advice_components/input_spec.rb [hunk 1]
```diff
@@ -112,6 +112,18 @@
...
+    end
+  end
+
+  context "when there is hint text and an error" do
+    let(:options) { { hint: "this is the hint text", error_message: "Enter your name" } }
+
+    it "adds aria-describedby to the input" do
+      expect(component.css("input").attribute("aria-describedby").value).to eq("example-input-error example-input-hint")
+    end
   end
```
Comment (davidrapson) [inline engine/spec/components/citizens_advice_components/input_spec.rb:127]: This is great. Could we add one more test to check that there's no `aria-describedby` when there's no error or hint? I reckon right now it'll output an empty attribute which is probably not what we want. Or maybe Rails is smart enough to work that out?

**Review by davidrapson — CHANGES_REQUESTED** [review]
This is really clear, great work. I reckon we could do with one more test: checking that the attribute is not present by default. And then this looks good to go!

File: engine/app/components/citizens_advice_components/input.rb [hunk 2]
```diff
@@ -86,10 +90,17 @@ def base_input_attributes
         value: value,
         required: required?,
         "aria-invalid": error?,
-        "aria-describedby": (error_id.to_s if error?)
+        "aria-describedby": described_by
       }
     end
 
+    def described_by
+      classes = []
```
Comment (davidrapson) [inline engine/app/components/citizens_advice_components/input.rb:98]: 💅 Nitpick: These are IDs not class names

File: engine/spec/components/citizens_advice_components/input_spec.rb (hunk 1 above)
Comment (elceebee) [reply engine/spec/components/citizens_advice_components/input_spec.rb:127]: Good call, I'll add one.

File: engine/app/components/citizens_advice_components/input.rb (hunk 2 above)
Comment (elceebee) [reply engine/app/components/citizens_advice_components/input.rb:98]: I'm so embarrassed.

File: styleguide/examples/textarea/hint.html [hunk 3]
```diff
@@ -8,8 +8,12 @@
     >
       Example input with hint
     </label>
-    <p class="cads-form-field__hint">This is the hint for the input</p>
+    <p class="cads-form-field__hint" id="example-input-with-hint-hint">
+      This is the hint for the input
+    </p>
     <textarea
+      aria-describedby="example-input-with-hint-hint"
+      aria-disabled="false"
```
Comment (davidrapson) [inline styleguide/examples/textarea/hint.html:16]: One more tiny thing sorry, where is this `aria-disabled` coming from? I know it's just in the examples but we shouldn't expect this to be here right?

**Review by davidrapson — APPROVED** [review]
One query about the `aria-disabled` that's turned up in the examples, but once that's removed merge away! 🚀

File: styleguide/examples/textarea/hint.html (hunk 3 above)
Comment (elceebee) [reply styleguide/examples/textarea/hint.html:16]: I was perhaps being too literal in interpreting the recommendations in the report. Here is the sample they give. It sounds like it isn't necessary. I'll remove it!

image

**Pull request merged**